deepchem/models/tensorgraph/tensor_graph.py

```diff
@@ -349,7 +349,11 @@ def predict_on_generator(self, generator, transformers=[], outputs=None):
 
       final_results = []
       for result_list in results:
-        final_results.append(np.concatenate(result_list, axis=0))
+        try:
```

Comment:
Since loss is an aggregated statistic it can't be concatenated (only a single entry per mini-batch).

We shouldn't fail hard in these cases -- but instead just return the results to the user chunked into mini-batches.